services/gitlab/gitlab-pipeline-status.service.js

```diff
@@ -0,0 +1,108 @@
+'use strict'
+
+const Joi = require('joi')
+const validate = require('../../lib/validate')
+const BaseSvgScrapingService = require('../base-svg-scraping')
+const { InvalidParameter, NotFound } = require('../errors')
+const { isPipelineStatus } = require('./gitlab-helpers')
+
+const badgeSchema = Joi.object({
+  message: Joi.alternatives()
+    .try([isPipelineStatus, Joi.equal('unknown')])
+    .required(),
+}).required()
+
+const queryParamsSchema = Joi.object({
+  // TODO This accepts URLs with query strings and fragments, which should be
+  // rejected.
+  gitlab_url: Joi.string().uri({ scheme: ['https'] }),
+}).required()
+
+module.exports = class GitlabPipelineStatus extends BaseSvgScrapingService {
+  static get category() {
+    return 'build'
+  }
+
+  static get route() {
+    return {
+      base: 'gitlab/pipeline',
+      format: '([^/]+)/([^/]+)(?:/([^/]+))?',
+      capture: ['user', 'repo', 'branch'],
+      // Trailing optional parameters don't work. The issue relates to the `.`
+      // separator before the extension.
+      // pattern: ':user/:repo/:branch?',
+      queryParams: ['gitlab_url'],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Gitlab pipeline status',
+        urlPattern: ':user/:repo',
+        exampleUrl: 'gitlab-org/gitlab-ce',
+        staticExample: this.render({ status: 'passed' }),
+      },
+      {
+        title: 'Gitlab pipeline status (branch)',
+        urlPattern: ':user/:repo/:branch',
+        exampleUrl: 'gitlab-org/gitlab-ce/master',
+        staticExample: this.render({ status: 'passed' }),
+      },
+      // TODO Find an example of a self-hosted gitlab.
```

Comment:
https://gitlab.gnome.org/ has lots of public repos and probably isn't going to go away in the near future. It might also be a good idea to have a service test that hits a self-hosted instance.